core/src/main/clojure/xtdb/metrics.clj

```diff
@@ -52,12 +54,34 @@
                #(.getFreeBytes ^Stats (get-stats))
                {:unit "bytes"})))
 
+(defn add-mem-cache-gauges [reg meter-name get-stats]
+  (add-cache-gauges reg meter-name get-stats)
+  (doto reg
+    (add-gauge (str meter-name ".metaSliceCount")
```

Comment:
suspect meta vs data should be a dimension here rather than part of the meter name (if that's exposed within micrometer?)

Reply:
I tried with:
- `Gauge` - can only be registered with the same name + same set of tags
- `Counter` - needs to be dealt with explicitly (ie `.increment()`)
- `FunctionCounter` - needs to be a monotonic increasing metric